SystemInformer/mainwnd.c

```diff
@@ -100,17 +100,84 @@ BOOLEAN PhMainWndInitialization(
         PhInitializeStringBuilder(&stringBuilder, 100);
         PhAppendStringBuilder2(&stringBuilder, PhApplicationName);
 
-        if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+        if (!PhGetIntegerSetting(L"EnableVerboseWindowTitle"))
         {
-            PhAppendStringBuilder2(&stringBuilder, L" [");
-            PhAppendStringBuilder(&stringBuilder, &currentUserName->sr);
-            PhAppendCharStringBuilder(&stringBuilder, L']');
-            if (KphIsConnected()) PhAppendCharStringBuilder(&stringBuilder, L'+');
-            PhDereferenceObject(currentUserName);
+            if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+            {
+                PhAppendStringBuilder2(&stringBuilder, L" [");
+                PhAppendStringBuilder(&stringBuilder, &currentUserName->sr);
+                PhAppendCharStringBuilder(&stringBuilder, L']');
+                if (KphIsConnected()) PhAppendCharStringBuilder(&stringBuilder, L'+');
+                PhDereferenceObject(currentUserName);
+            }
+
+            if (PhGetOwnTokenAttributes().ElevationType == TokenElevationTypeFull)
+                PhAppendStringBuilder2(&stringBuilder, L" (Administrator)");
         }
+        else
+        {
+            DWORD myPID = GetCurrentProcessId();
+            WCHAR myPIDStrArr[PH_INT32_STR_LEN_1];
+            PhPrintUInt32(myPIDStrArr, (ULONG)myPID);
+            PhAppendStringBuilder2(&stringBuilder, L" ");
+            PhAppendStringBuilder2(&stringBuilder, myPIDStrArr);
+            PhAppendStringBuilder2(&stringBuilder, L" ");
+
+            BOOL isRunningAsSystem = FALSE;
+
+            if (currentUserName = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+            {
+                PPH_STRING strSystemUsername = PhCreateString(L"NT Authority\\System");
+                isRunningAsSystem = (PhCompareString(currentUserName, strSystemUsername, TRUE) == 0);
+                PhDereferenceObject(strSystemUsername);
+
+                if (KphIsConnected())
+                    PhAppendStringBuilder2(&stringBuilder, L" <Driver-Connected>");
+                else
+                    PhAppendStringBuilder2(&stringBuilder, L" <No-Driver>");
+
+                PhDereferenceObject(currentUserName);
+            }
+
+            // Session ID\WindowStation\Desktop ...
```

Comment:
Clever way to format this info. I think you should clean up and combine it into a format string like I suggested above.